File: py/1941.check-if-all-characters-have-equal-number-of-occurrences.py
#
# @lc app=leetcode id=1941 lang=python3
#
# [1941] Check if All Characters Have Equal Number of Occurrences
#
from collections import Counter
# @lc code=start


class Solution:
    def areOccurrencesEqual(self, s: str) -> bool:
        stat = Counter()
        for c in s:
            stat[c] += 1
        num = next(iter(stat.values()))
        for val in stat.values():
            if val != num:
                return False
        return True
# @lc code=end

File: cpp/1941.check-if-all-characters-have-equal-number-of-occurrences.cpp
/*
 * @lc app=leetcode id=1941 lang=cpp
 *
 * [1941] Check if All Characters Have Equal Number of Occurrences
 */
#include <string>
using namespace std;
// @lc code=start
class Solution
{
public:
    bool areOccurrencesEqual(string s)
    {
        string count(26, 0);
        for (auto c : s)
        {
            ++count[c - 'a'];
        }
        int num = 0;
        for (auto n : count)
        {
            if (n != 0)
            {
                if (num == 0)
                {
                    num = n;
                }
                else
                {
                    if (num != n)
                    {
                        return false;
                    }
                }
            }
        }
        return true;
    }
};
// @lc code=end
